src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -3,13 +3,20 @@
 import java.time.LocalDate;
 import java.util.List;
 import java.util.Optional;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import wolox.training.models.User;
 
 public interface UserRepository extends CrudRepository<User, Long> {
 
 	Optional<User> findByUsername(String username);
 
-	List<User> findByBirthdateBetweenAndNameContainingIgnoreCase(LocalDate startDate, LocalDate endDate,String name);
+	@Query("SELECT u FROM User u WHERE (CAST(:dtCancelInitial AS java.time.LocalDateTime) IS NULL OR (u.birthdate >= "
```

Comment:
I think all the values must be validated, try to valid all the values (startDate, endDate and name)